pkg/pool-hooks/contracts/StableSurgePoolFactory.sol

```diff
@@ -25,17 +25,16 @@ contract StableSurgePoolFactory is IPoolVersion, BasePoolFactory, Version {
     string private _poolVersion;
 
     constructor(
-        IVault vault,
+        StableSurgeHook stableSurgeHook,
```

Comment:
Just a thought: should we have an `IStableSurgeHook` interface for this hook and use it here? (e.g., it would declare `getDefaultMaxSurgeFeePercentage`, `getDefaultSurgeThresholdPercentage`, etc.). Would that make things easier for integrators?

Reply:
I've created the interface, but here we need to get the vault as well so we need the whole contract.

Reply:
We could also use the interface here and cast the contract as `VaultGuard` where we need to get the vault, but using the whole contract here sounds simpler, so I'd keep the code as is.